go/vt/wrangler/reparent.go

```diff
@@ -406,52 +408,191 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 		return fmt.Errorf("the shard has no master, use EmergencyReparentShard")
 	}
 
-	// create a new context for the short running remote operations
-	remoteCtx, remoteCancel := context.WithTimeout(ctx, *topo.RemoteOperationTimeout)
-	defer remoteCancel()
+	// Find the current master (if any) based on the tablet states. We no longer
+	// trust the shard record for this, because it is updated asynchronously.
+	currentMaster := wr.findCurrentMaster(tabletMap)
 
 	var reparentJournalPos string
 
-	if topoproto.TabletAliasEqual(shardInfo.MasterAlias, masterElectTabletAlias) {
-		// If the master is already the one we want, we just try to fix replicas (below).
-		rp, err := wr.tmc.MasterPosition(remoteCtx, masterElectTabletInfo.Tablet)
+	if currentMaster == nil {
```

Comment:
Since PRS is now handling the cases of no master / multi-master, that means situations where ERS is required should become rare to nonexistent. Let us make sure to document that in the eventual PR for merging upstream.

Reply:
Good point. To summarize, my goal for PRS is that eventually it should be able to fix almost any problem as long as:

1. All tablets are reachable, so we can check global state.
   AND
2. The global replication state (relative positions) is consistent and compatible with making the chosen tablet the master.

You should then only need ERS in the following cases:
1. The current master is unreachable.
   OR
2. The relative replication positions have become inconsistent (e.g. alternative futures).
   OR
3. It's unclear who the current master is, and some tablets are unreachable, which means we can't be sure if the global state is consistent.